tests/test_mode_manager.py

```
@@ -0,0 +1,64 @@
+    assert mm.is_build_mode()
+    assert "ASK" in mm.set_mode("ask")
+    assert not mm.is_build_mode()
+
+def test_permission_flow(tmp_path):
+    mm = ModeManager()
+    mm.set_mode("build")
+    file_path = tmp_path / "test.txt"
+    file_path.write_text("hello")
+    change = FileChange(str(file_path), "hello", "world", "edit")
+    mm.add_pending_change(change)
+    diff = change.get_diff()
```

Ruff (0.12.2)

Local variable `diff` is assigned to but never used

Remove assignment to unused variable `diff`

(F841)